### PR TITLE
Add workflows for Building and Deploying Docker Images

### PR DESCRIPTION
**Pull request by FinnIckler**
In Preparation for #8314 I added two new workflows.
On Release:
- Builds new Images and marks them as latest for staging and prod
- Pushing an image triggers the deployment pipeline for prod
- Sidekiq image is updated and deployed
- Staging image is updated and deployed

On PR Comment:
- Allows deploying PRs to staging by writing '@thewca-bot deploy staging' 
- Replaces Main and Sidekiq Staging images if triggered

We could think of another way of deploying staging images, but I think this could work well. This almost makes the deploy repository redundant, depending on if we want to add a separate workflow for updating documents.

### CODE REVIEW

**Comment by gregorbg** [comment]
> This almost makes the deploy repository redundant, depending on if we want to add a separate workflow for updating documents.

I think the documents repository is self-sufficient and they don't manage their deployments over `thewca/deploy`. But obviously they still need to maintain a way to update their documents independent of the main code, so you probably need a PR over there to adjust the commands to the new infrastructure.

Then the only remaining GH Workflow in the deploy repository is the one for Regulations. WRC definitely needs it, and the reasons why it lives in the deploy repository are:

1. It already had a lot of Secrets installed so it was more convenient to configure there, rather than moving it to the regulations repository.
2. We were not sure whether having a workflow that can be manually triggered in a public repository could be dangerous. External people defintely don't have permission to trigger workflows, but (I think) they do have permissions to view workflow logs. Could that be problematic in any way if we moved the Regulations workflow to the (public!) Regulations repository?

File: .github/workflows/deploy-on-release.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          tags: |
+            ${{ steps.login-ecr.outputs.registry }}/wca-on-rails:latest
+            ${{ steps.login-ecr.outputs.registry }}/wca-on-rails:staging
+            ${{ steps.login-ecr.outputs.registry }}/wca-on-rails:${{ github.event.release.tag_name }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+        # Replace the old sidekiq image with the new one
+      - name: Deploy Sidekiq
+        run: |
+          aws ecs update-service --cluster wca-on-rails --service wca-on-rails-auxiliary-services --force-new-deployment
```
Comment (gregorbg) [inline .github/workflows/deploy-on-release.yml:48]: Here it seems like you're deploying the _production_ Sidekiq but...

File: .github/workflows/deploy-on-release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        # There is no pipeline for staging so we manually force to update the image
+      - name: Deploy staging
+        run: |
+          aws ecs update-service --cluster wca-on-rails --service wca-on-rails-staging --force-new-deployment
```
Comment (gregorbg) [inline .github/workflows/deploy-on-release.yml:52]: ...here in turn you're deploying the _staging_ image. What about production?

File: .github/workflows/staging-deployment-from-prs.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        uses: docker/build-push-action@v4
+        with:
+          context: ./WcaOnRails
+          push: true
+          file: Dockerfile
+          tags: |
+            ${{ steps.login-ecr.outputs.registry }}/wca-on-rails:staging
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Deploy staging if triggered
```
Comment (gregorbg) [inline .github/workflows/staging-deployment-from-prs.yml:66]: You've built Sidekiq but it seems like you're never deploying it.

Comment (FinnIckler) [reply .github/workflows/staging-deployment-from-prs.yml:66]: On staging there is only one service, so forcing `wca-on-rails-staging ` to update will deploy both the rails and the sidekiq image

File: .github/workflows/deploy-on-release.yml (hunk 2 above)
Comment (FinnIckler) [reply .github/workflows/deploy-on-release.yml:52]: Production is deployed automatically on Docker image push through a blue-green deployment Pipleine

**Pull request merged**